### PR TITLE
Add calendar providers

### PR DESCRIPTION
**Pull request by joklek**
Implemented ability to add calendar providers, so for example we might add a Google Calendar event provider, to check when account is taken. Querying when account is taken for trips has been migrated to `AccountTripCalendarProvider.java`. For frontend devs to differiantiate, the result of such provider has a type

Also added DTOs for normalized behavior for the api

### CODE REVIEW

**Comment by joklek** [comment]
Reikėjo atskirą branchą susikurti DTO pakeitimams, srr my dudes

File: src/main/java/com/javainiaisuzspringom/tripperis/domain/ChecklistItem.java [hunk 1]
```diff
@@ -27,4 +30,14 @@
     @ManyToOne(cascade = CascadeType.ALL)
     @JoinColumn(name = "trip_id")
     private Trip trip;
+
+    public ChecklistItemDTO convertToDTO() {
+        ChecklistItemDTO item = new ChecklistItemDTO();
+
+//        item.setId(this.getId());
```
Comment (Adomas202) [inline src/main/java/com/javainiaisuzspringom/tripperis/domain/ChecklistItem.java]: Girdi, gal trini visus komentarus 💯💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯

File: src/main/java/com/javainiaisuzspringom/tripperis/services/TripStepService.java [hunk 2]
```diff
@@ -1,27 +1,66 @@
...
+                });
+            }
+        }
+        return convertToEntity(dto);
+    }
+
+    private TripStep convertToEntity(TripStepDTO dto) {
+        TripStep tripStep = new TripStep();
+
+//        tripStep.setId(this.getId());
```
Comment (Adomas202) [inline src/main/java/com/javainiaisuzspringom/tripperis/services/TripStepService.java]: 😿

File: src/main/java/com/javainiaisuzspringom/tripperis/services/TripService.java [hunk 3]
```diff
@@ -23,47 +29,95 @@
...
+                });
+            }
+        }
+        return convertToEntity(dto);
+    }
+
+    private Trip convertToEntity(TripDTO dto) {
+        Trip trip = new Trip();
+
+//        trip.setId(dto.getId());
```
Comment (Adomas202) [inline src/main/java/com/javainiaisuzspringom/tripperis/services/TripService.java]: 😡

File: src/main/java/com/javainiaisuzspringom/tripperis/services/StatusCodeService.java [hunk 4]
```diff
@@ -16,12 +19,45 @@
...
+                });
+            }
+        }
+        return convertToEntity(dto);
+    }
+
+    private StatusCode convertToEntity(StatusCodeDTO dto) {
+        StatusCode statusCode = new StatusCode();
+
+//        statusCode.setId(this.getId());
```
Comment (Adomas202) [inline src/main/java/com/javainiaisuzspringom/tripperis/services/StatusCodeService.java]: 😠

**Pull request merged**